app/modules/munkiinfo/scripts/munkiinfo.py

```diff
@@ -106,9 +128,13 @@ def munkiinfo_report():
     if AppleCatalogURL == 'None':
         AppleCatalogURL = ''
 
+    middleware_name, middleware_version = middleware_checks()
+
     report = {
         'AppleCatalogURL': AppleCatalogURL,
         'munkiprotocol': munkiprotocol,
+        'MiddlewareName': middleware_name,
+        'MiddlewareVersion': middleware_version,
     }
```

Comment:
This creates a string type when the s3 middleware is present
```xml
		<key>MiddlewareName</key>
		<string>middleware_s3</string>
		<key>MiddlewareVersion</key>
		<array>
		</array>
```

However it creates a double array when no middleware is present
```xml
		<key>MiddlewareName</key>
		<array>
		</array>
		<key>MiddlewareVersion</key>
		<array>
		</array>
```

This needs to be corrected to be an array of strings all of the time.